src/main/java/Duke.java

```diff
@@ -1,10 +1,181 @@
+
+        // Active Chat
+        ActiveChat();
+
+        // Goodbye Message
+        PrintGoodbyeMessage();
+    }
+
+    private static void ActiveChat() {
+        boolean isBye = false;
+        boolean isEmptyList = true;
+        String input;
+        Scanner in = new Scanner(System.in);
+        int counter = 0;
+        Task[] list = new Task[100];
+
+        while(!isBye){
+            //store input
+            input = in.nextLine();
+            //check if input is empty
+            CheckIsEmptyList(list);
+            //process input
+            if (input.equals("bye")){ //check if bye
+                isBye = true;
+            } else if (input.equals("list")) { //check if list
+                ProcessList(counter, list);
+            } else if (input.contains("done") ) { //check if done
+                ProcessDone(input, list);
+            } else {
+                counter = ProcessTasks(input, counter, list); //process tasks
+            }
+        }
+    }
+
+    private static void CheckIsEmptyList(Task[] list) {
+        boolean isEmptyList;
+        if(list[0] != null) {
+            isEmptyList = false;
+        }
+    }
+
+    private static void ProcessList(int counter, Task[] list) {
+        if (list[0] != null){
+            PrintListMessage(counter, list);
+        } else {
+            PrintListButEmptyMessage();
+        }
+    }
+
+    private static void ProcessDone(String input, Task[] list) {
+        int donePos = input.indexOf("done");
+        if (list[0] != null) {
+            if (input.length() < donePos + 5) {
+                PrintDoneButNotSpecificMessage(); //when input contains done but no number
+            } else {
+                ProcessDone(input, list, donePos); //when input contains done and specified number
+            }
+        } else {
+            PrintDoneButEmptyMessage(); //when input contains done but list is empty
+        }
+    }
+
+    private static void ProcessDone(String input, Task[] list, int donePos) {
+        String itemNumDone = input.substring(donePos + 5, donePos + 6);
+        int itemNum = Integer.parseInt(itemNumDone);
+        list[itemNum - 1].setDone();
+        PrintDoneMessage(list, itemNum);
+    }
+
+    private static int ProcessTasks(String input, int counter, Task[] list) {
+        if (input.contains("todo")) {
+            String description = input.substring(5);
+            ToDo newTask = new ToDo(description);
+            list[counter] = newTask;
+            counter += 1;
+            PrintAddedTaskMessage(newTask, counter);
+        } else if (input.contains("deadline")) {
+            int donePos = input.indexOf("/"); //finds pos of '/'
+            String description = input.substring(9,donePos);
+            String date = input.substring(donePos + 4);
+            Deadline newTask = new Deadline(description,date);
+            list[counter] = newTask;
+            counter += 1;
+            PrintAddedTaskMessage(newTask, counter);
```

Comment:
It might be cleaner to refactor this into a separate method. 
One way is to have additional methods in the Duke class that handles the different Task subclasses. 
Another is to create a separate class `TaskManager` that stores the array of Tasks and has functions to add the different Task subclasses to keep the main Duke class clean. __(I think this will be the best approach)__
Another is to overload the subclass constructors to take in `String[]` or a similar data structure.